xlsxwriter/worksheet.py

```diff
@@ -52,6 +62,59 @@
 )
 from .xmlwriter import XMLwriter
 
+if TYPE_CHECKING:
+    from typing_extensions import Concatenate, ParamSpec, Protocol, overload
+
+    _ReturnT_co = TypeVar("_ReturnT_co", covariant=True)
+    _P = ParamSpec("_P")
```

Comment:
Based on your link to the  pylint restrictions on naming I would suggest calling these `ReturnTypeT` and `P` (since the only other option `_P` is worse).